Detectors/TPC/base/include/TPCBase/CalArray.h

```diff
@@ -165,6 +198,71 @@ inline const CalArray<T>& CalArray<T>::operator+= (const CalArray& other)
 
 //______________________________________________________________________________
 template <class T>
+inline const CalArray<T>& CalArray<T>::operator-= (const CalArray& other)
+{
+  if ( !((mPadSubset == other.mPadSubset) && (mPadSubsetNumber == other.mPadSubsetNumber) ) ){
+    LOG(ERROR) << "You are trying to operate on incompatible objects: Pad subset type and number must be the same on both objects"
+               << FairLogger::endl;
+    return *this;
+  }
+  for (size_t i=0; i<mData.size(); ++i) {
+    mData[i] -= other.getValue(i);
+  }
+  return *this;
+}
+
+//______________________________________________________________________________
+template <class t>
+inline const CalArray<t>& CalArray<t>::operator*= (const CalArray& other)
+{
+  if ( !((mPadSubset == other.mPadSubset) && (mPadSubsetNumber == other.mPadSubsetNumber) ) ){
+    LOG(ERROR) << "pad subste type of the objects it not compatible" 
+               << FairLogger::endl;
+    return *this;
+  }
```

Comment:
Since two calarrays of different subsetnumbers not compatible, you could consider making the subset a template argument to calarray. Then the compiler would forbid operations between incompatible arrays.

Reply:
Yes, but this would bring other problems. I think I considered this at some point. I will check this in future, but propose to keep this out for the moment.